airflow/decorators/task_group.py

```diff
@@ -106,12 +107,16 @@ def _make_task_group(self, **kwargs) -> MappedTaskGroup:
         return tg
 
     def partial(self, **kwargs) -> "MappedTaskGroupDecorator[R]":
-        if self.partial_kwargs:
-            raise RuntimeError("Already a partial task group")
+        duplicated_keys = [k for k in kwargs if k in self.partial_kwargs]
+        if len(duplicated_keys):
+            raise ValueError(f"Cannot overwrite partial argument: {duplicated_keys[0]!r}")
+        elif duplicated_keys:
+            joined = ", ".join(repr(k) for k in duplicated_keys)
+            raise ValueError(f"Cannot overwrite partial arguments: {joined}")
```

Comment:
or alternatively... 

```suggestion
duplicated_keys = [k for k in kwargs if k in self.partial_kwargs]
if duplicated_keys:
    raise ValueError(f"Cannot overwrite partial arguments: {duplicated_keys!r}")
```

personally i wouldn't sweat the grammar but different strokes :)